File: LinkedList/LinkedList.cpp
#include <bits/stdc++.h>
using namespace std;

class Node
{
    public :
    int data;
    Node *next;
       
};

class LinkedList
{
  public:
    Node *head=NULL;
    Node *p,*q;
     int count=0;
    void insert()
    {
      int ele;
      cout<<"Enter the element to insert:";
      cin>>ele;
      if(head==NULL)
      { 
          p=new Node();
          p->data=ele;
          p->next=NULL;
          head=p;
      }
      else 
      {
          p->next=new Node();
          p=p->next;
          p->data=ele;
          p->next=NULL;
      }
      count+=1;
    }
 
     void pop()
     {
         q=head;
         if(q==NULL)
         {
             cout<<"The list is empty"<<endl;
             return;
         }
         if(q->next==NULL)
         {
             cout<<"The element "<<q->data<<" is successfully popped"<<endl;
             head=NULL;
             free(q); 
             count-=1;
             return;
         }
         while(q->next->next!=NULL)
         {
            q=q->next;
         }
         q->next=NULL;
         cout<<"The element "<<p->data<<" is successfully popped"<<endl;
         free(p);
         p=q;
         count-=1;
     }

     void presentsize()
     {
         cout<<"The size of linked list is "<<count<<endl;
     }

    void printList()
    {
       q=head;
       if(q==NULL)
       {
           cout<<"The list is empty"<<endl;
           return;
       }
       while(q!=NULL)
       {
           cout<<q->data<<" ";
           q=q->next;
       }
       cout<<endl;
    }
};
int main()
{
    LinkedList ll=LinkedList();
    while(true)
    {
        cout<<"1.Append    2.pop    3.printList   4.presentsize   5.exit"<<endl;
        cout<<"Enter your choice: ";
        int choice;
        cin>>choice;
       
       if(choice==1)
       {
           ll.insert();
       }
       else if(choice==2)
       {
          ll.pop();
       }
       else if(choice==3)
       {
           ll.printList();
       }
       else if(choice==4)
       {
          ll.presentsize();
       }
       else
       {
           break;
       }
    }


}

File: LinkedList/LinkedList.java
import java.util.Scanner;

class Node
{
    int data;
    Node next;
    public Node(int data)
    {
      this.data=data;
      this.next=null;
    }
}

class LinkedList
{
    Node head=null;
    Node p=null;
    Node q=null;
    int size=0;
     void insert(int ele)
    {
       if(head==null)
       {
           p=new Node(ele);
           head=p;
       }
       else
       {
         p.next=new Node(ele);
         p=p.next;
       }
       size+=1;
    }

    void pop()
    {
        q=head;
        if(q==null)
        {
            System.out.println("linked list is empty");
            return;
        }
        if(q.next==null)
        {
            System.out.println("Element "+q.data+" is successfully popped");
           head=null;
           p=head;
           size-=1;
           return;
        }
        while(q.next!=p)
        {
            q=q.next;
        }
        System.out.println("Element "+p.data+" successfully popped");
        q.next=null;
        p=q;
        size-=1;
    }
    void printList()
    {
        if(head==null)
        {
            System.out.println("The List is Empty");
            return;
        }
        System.out.print("The elements in Linked List are: ");
        q=head;
        while(q!=null)
        {
            System.out.print(q.data+" ");
            q=q.next;
        }
        System.out.println();
    }
    public static void main(String[] args)
    {
      int ele;
      LinkedList ll=new LinkedList();
      Scanner sc=new Scanner(System.in);
      while(true)
      {
          System.out.println("1.Insert   2.Pop   3.PrintList   4.presentsize   5.exit");
          System.out.print("Enter your choice: ");
          int choice=sc.nextInt();
          if(choice==1)
          {
              System.out.print("Enter the element to insert: ");
              ele=sc.nextInt();
              ll.insert(ele);
          }
          else if(choice==2)
          {
              ll.pop();
          }
          else if(choice==3)
          {
              ll.printList();
          }
          else if(choice==4)
          {
              System.out.println("The size of Linkedlist is "+ll.size);
          }
          else
          {
             sc.close();
             break;
          }
      }
    }
}
